apps/nova/src/components/TutorProfile/Ratings.tsx

```diff
@@ -1,74 +1,107 @@
+import React, { useCallback } from "react";
+import { LoadingTutorInfo } from "@/components/TutorProfile/LoadingTutorInfo";
+import { ratingOrganizer } from "@/lib/ratings";
+import { useUser } from "@litespace/headless/context/user";
+import {
+  TutorRatingCard,
+  TutorRatingCardGroup,
+} from "@litespace/luna/TutorFeedback";
+import { Typography } from "@litespace/luna/Typography";
+import { useFormatMessage } from "@litespace/luna/hooks/intl";
+import Star from "@litespace/assets/Star";
+import NewTutor from "@litespace/assets/NewTutor";
+import { Button, ButtonSize } from "@litespace/luna/Button";
+import { isEmpty } from "lodash";
+import { TutorLoadingError } from "@/components/TutorProfile/TutorLoadingError";
 
-const Ratings: React.FC<{
-  query: UseQueryResult<IRating.FindRateeRatingsApiResponse, Error>;
-}> = ({ query }) => {
-  const [editRating, setEditRating] = useState<IRating.Populated | null>(null);
-  const [deleteRating, setDeleteRating] = useState<{
-    ratingId: number;
-    tutorId: number;
-  } | null>(null);
-
-  const onEditRating = useCallback(
-    (rating: IRating.Populated) => setEditRating(rating),
-    []
+const NoTutorRatings: React.FC<{ tutorName: string | null }> = ({
+  tutorName,
+}) => {
+  const intl = useFormatMessage();
+  return (
+    <div className="flex flex-col mt-20 gap-14 items-center justify-center">
+      <NewTutor />
+      <Typography
+        element="h4"
+        weight="bold"
+        className="text-natural-950 text-center mb-20"
+      >
+        {intl("tutor.profile.first-rating", { tutor: tutorName })}
+      </Typography>
+    </div>
   );
+};
 
-  const onDeleteRating = useCallback(
-    (payload: { ratingId: number; tutorId: number }) =>
-      setDeleteRating(payload),
-    []
-  );
+const Ratings: React.FC<{ id: number; tutorName: string | null }> = ({
+  id,
+  tutorName,
+}) => {
+  const { user } = useUser();
+  const intl = useFormatMessage();
+  const ratingsQuery = useFindTutorRatings(id);
 
-  const onClose = useCallback(() => {
-    setEditRating(null);
-    setDeleteRating(null);
-  }, []);
+  const refetchRatings = useCallback(() => {
+    ratingsQuery.refetch();
+  }, [ratingsQuery]);
 
-  const timeline = useMemo((): TimelineItem[] => {
-    if (!query.data) return [];
+  if (ratingsQuery.isLoading || ratingsQuery.isPending)
+    return <LoadingTutorInfo />;
 
-    return query.data.list.map((rating) => ({
-      id: rating.id,
-      children: (
-        <Rating
-          rating={rating}
-          onEdit={onEditRating}
-          onDelete={onDeleteRating}
-        />
-      ),
-      icon: <Star />,
-    }));
-  }, [onDeleteRating, onEditRating, query.data]);
+  if (ratingsQuery.error || !user)
+    return <TutorLoadingError refetchRatings={refetchRatings} />;
 
-  if (query.isLoading) return <Loading className="h-60" />;
-  if (query.isError) return <h1>Error</h1>;
-  if (!query.data) return null;
+  const ratings = ratingOrganizer(ratingsQuery.data.list, user.id);
```

Comment:
For function names, it is better to use verbs. In this case it is better to name it `orgainzeRatings` or `groupRatings`.